pkg/collector/python/loader.go

```diff
@@ -239,11 +239,12 @@ func expvarPy3Warnings() interface{} {
 // reportPy3Warnings runs the a7 linter and exports the result in both expvar
 // and the aggregator (as extra series)
 func reportPy3Warnings(checkName string, checkFilePath string) {
-	statsLock.Lock()
-	defer statsLock.Unlock()
 
 	// check if the check has already been linted
-	if _, found := py3Warnings[checkName]; found {
+	statsLock.RLock()
```

Comment:
The function `expvarPy3Warnings` seems to never be threadsafe:  https://github.com/DataDog/datadog-agent/pull/4966/files#diff-5d538e4107a2ef1eb9323682d9dad905R232-R237.
With your changes, you may increase the likelihood of having a race condition IMO.

Reply:
The sole purpose of `expvarPy3Warnings` is to provide expvar, and its output now depends on the amount of .py code to lint (and the time taken to do so), in that extend it is a race condition because the function output depends on external unpredictable factor. So to make it consistent we would have to wait for the linter to finish, that was the case before this PR but it logically freezes any caller of `expvarPy3Warnings` during  the linting. And the purpose of this PR was to address the `agent status` freezing issue 🤔

Reply:
I added a backlog task to deal with `expvarPy3Warnings` thread-safety possible issue (and other similar issue).

Reply:
I put the PR on hold pending the resolution of the aforementioned issue.

Reply:
#5151 resolved the thread safety issue.